channeldb/payment_control.go

```diff
@@ -6,11 +6,18 @@ import (
 	"errors"
 	"fmt"
 	"io"
+	"sync"
 
 	"github.com/lightningnetwork/lnd/kvdb"
 	"github.com/lightningnetwork/lnd/lntypes"
 )
 
+const (
+	// paymentSeqBlockSize is the block size used when we batch allocate
+	// payment sequences for future payments.
+	paymentSeqBlockSize = 1000
```

Comment:
👍 I really like this approach as we have several places requesting sequence generation. We could apply it to all the sequencers and the gain would be substantial!